spec/rails_helper.rb

```diff
@@ -82,7 +92,48 @@
 Capybara.default_driver = :rack_test # This is a faster driver
 Capybara.javascript_driver = :selenium_chrome_headless_sandboxless # This is slower
 
+Capybara::Screenshot.register_driver(:selenium_chrome_headless_sandboxless) do |driver, path|
+  driver.browser.save_screenshot(path)
+end
+
+Capybara::Screenshot.autosave_on_failure = false
+
+# Save CircleCI artifacts
+
+def save_timestamped_page_and_screenshot(page, meta)
+  filename = File.basename(meta[:file_path])
+  line_number = meta[:line_number]
+
+  time_now = Time.now
+  timestamp = "#{time_now.strftime('%Y-%m-%d-%H-%M-%S.')}#{'%03d' % (time_now.usec/1000).to_i}"
+
+  screenshot_name = "screenshot-#{filename}-#{line_number}-#{timestamp}.png"
+  screenshot_path = "#{Rails.root.join('tmp/capybara')}/#{screenshot_name}"
+  page.save_screenshot(screenshot_path)
+
+  page_name = "html-#{filename}-#{line_number}-#{timestamp}.html"
+  page_path = "#{Rails.root.join('tmp/capybara')}/#{page_name}"
+  page.save_page(page_path)
+
+  puts "\n  Screenshot: tmp/capybara/#{screenshot_name}"
+  puts "  HTML: tmp/capybara/#{page_name}"
+end
+
 RSpec.configure do |config|
+  config.before(:each, type: :system) do
+    driven_by :rack_test
+  end
+
+  config.before(:each, type: :system, js: true) do
+    driven_by :selenium_chrome_headless_sandboxless #
+
+    if ENV['IN_DOCKER'].present?
+      Capybara.server_host = '0.0.0.0'
+      Capybara.server_port = 3010
+      # renaming container because the app domain is taken by google(gTLD)
+      Capybara.app_host = 'http://essi_app:3010'
```

Comment:
ah! I take it all back, that makes sense.  maybe just call it "essi" everywhere instead of essi_app

Reply:
if this config is done above (where it is in hyku and where you deleted it) it should work just fine. I don't think you need to set it every single time like this.